### ISSUE TITLE
Make fields of wrapped structures directly accessible

### ISSUE THREAD

**Issue by Seelengrab** [JuliaGPU/Vulkan.jl#18]
E.g. `Extent2D.width` or `Extent2D.height` instead of having to do `Extent2D.vks.width` and `Extent2D.vks.height`. I seem to remember that other types exposed the fields of the wrapped type directly.

**Comment by serenity4** [comment]
Only the structures that are only returned from Vulkan but never required as input have all fields exposed. All the others only wrap around the `vks` field, because they need an identical layout with that of the C API.

**Comment by serenity4** [comment]
In the case of `Extent2D`, `Offset2D` etc. the layout would be identical to the wrapped version, but in general it is not the case, particularly when there are pointer fields.

**Comment by Seelengrab** [comment]
Hmm, I see - I guess they could be passed through automatically by customizing `getproperty` for those types? So the relevant field symbols from the `Vk` version could just be passed through to `getproperty` here, allowing `Extent2D.width` and friends to work.

**Comment by serenity4** [comment]
Great idea! I'll add that on my list. So we would forward all `getproperty` calls to the `vks` field, except for the "real" fields of the structure (including `vks` itself).

**Comment by Seelengrab** [comment]
yep, precisely - it's a somewhat common pattern for wrapping C-ABI structs in proper julia objects, as far as I'm aware.

**Comment by serenity4** [comment]
The main issue with the wrapper structs being the lack of introspection, I took a different approach that consists of generating more high-level structures that expose all their fields with idiomatic Julia types, and usable in API calls. The "old" ones are still present, being slightly faster to use, prefixed with an underscore. For `Extent2D` mentioned in this issue, the fields `extent` and `width` are accessible directly, but not for `_Extent2D` which represents the old version. See [the documentation](https://juliagpu.github.io/Vulkan.jl/dev/tutorial/types/#Structures) for more details about these two kinds of structures and how they are related.

I released a new version which contains these changes. See the [CHANGELOG](https://github.com/JuliaGPU/Vulkan.jl/blob/master/CHANGELOG.md) for more details, notably about breaking changes.
